docs/metrics.md

```diff
@@ -28,6 +28,77 @@ Accessing metrics requires Prometheus and Grafana installed. Follow the
 [instructions to install Prometheus and Grafana](https://github.com/knative/docs/blob/master/docs/serving/installing-logging-metrics-traces.md)
 in namespace `knative-monitoring`.
 
+
+## Prometheus Collection
+> _All commands assume root of repo._ 
+
+1. Enable Knatives install of Prometheus to scrape Knative with GCP, run the
+   following:
+
+   ```shell
+   kubectl get configmap -n  knative-monitoring prometheus-scrape-config -oyaml > tmp.prometheus-scrape-config.yaml
+   sed -e 's/^/    /' config/monitoring/metrics/prometheus/100-prometheus-scrape-kn-eventing.yaml > tmp.prometheus-scrape-kn-eventing.yaml
+   sed -e '/    scrape_configs:/r tmp.prometheus-scrape-kn-eventing.yaml' tmp.prometheus-scrape-config.yaml \
+     | kubectl apply -f -
+   ```
+
+2. To verify, run the following to show the diff between what the resource was
+   original and is now (_note_: k8s will update `metadata.annotations`):
+
+   ```shell
+   kubectl get configmap -n  knative-monitoring prometheus-scrape-config -oyaml \
+     | diff - tmp.prometheus-scrape-config.yaml
+   ```
+
+   Or, to just see our changes (without `metadata.annotations`) run:
+
+   ```shell
+   CHANGED_LINES=$(echo $(cat config/monitoring/metrics/prometheus/100-prometheus-scrape-kn-eventing.yaml | wc -l) + 1 | bc)
+   kubectl get configmap -n  knative-monitoring prometheus-scrape-config -oyaml \
+     | diff - tmp.prometheus-scrape-config.yaml \
+     | head -n $CHANGED_LINES
+   ```
+
+3. Restart Prometheus
+
+   To pick up this new config, the pods that run Prometheus need to be
+   restarted, run:
+
+   ```shell
+   kubectl delete pods -n knative-monitoring prometheus-system-0 prometheus-system-1
+   ```
+
+   And they will come back:
+
+   ```shell
+   $ kubectl get pods -n knative-monitoring
+   NAME                                 READY   STATUS    RESTARTS   AGE
+   grafana-d7478555c-8qgf7              1/1     Running   0          22h
+   kube-state-metrics-765d876c6-z7dfn   4/4     Running   0          22h
+   node-exporter-5m9cz                  2/2     Running   0          22h
+   node-exporter-z59gz                  2/2     Running   0          22h
+   prometheus-system-0                  1/1     Running   0          32s
+   prometheus-system-1                  1/1     Running   0          36s
+   ```
+
+4. To remove the temp files, run:
+
+   ```shell
+   rm tmp.prometheus-scrape-kn-eventing.yaml
+   rm tmp.prometheus-scrape-config.yaml
+   ```
+
```

Comment:
I think you are going to add a new scrap job `eventing-controller` to Prometheus scrap configuration. But the instructions look very complicated. Do you want to try "kubectl patch configmap prometheus-scrape-config ..." ? 

An example is that how we add the eventing dashboard to the configmap `grafana-dashboard-definition-knative` :
`kubectl patch configmap grafana-dashboard-definition-knative --patch "$(cat config/monitoring/metrics/grafana/100-grafana-dash-eventing.yaml)" -n knative-monitoring`

Reply:
@daisy-ycguo The problem here is the scrapping config needs to be a part of the same file([key](https://github.com/knative/serving/blob/b105e59466e46a1d0aeadbce7e950733a1e09a9c/config/monitoring/metrics/prometheus/100-prometheus-scrape-config.yaml#L24)) in the configmap and we can't patch like the example you posted. Hence this complicated way of doing things.

Reply:
@sayanh Thank you for clarify. I understand now, but still feel complicated. I wonder if `kubectl patch configmap --type json ...` could help. Here is a [reference](http://erosb.github.io/post/json-patch-vs-merge-patch/) to `json patch`